File: 04-Socket-UDP/esempi/01.01-unicast-java/UdpReceiver.java
/**
 * UdpReceiver.java
 * Esempio di ricevitore UDP in Java
 * Riceve un messaggio UDP e lo stampa a schermo
 * Versione: 1.0
 * Data: 18/10/25
 * Autore: Filippo Bilardo
 * 
 * javac UdpReceiver.java && java UdpReceiver
 */
import java.net.DatagramPacket;
import java.net.DatagramSocket;

public class UdpReceiver {
    public static void main(String[] args) {
        try {
            DatagramSocket socket = new DatagramSocket(9876);
            byte[] buffer = new byte[1024];
            DatagramPacket packet = new DatagramPacket(buffer, buffer.length);

            System.out.println("In attesa di messaggi...");
            socket.receive(packet);
            String message = new String(packet.getData(), 0, packet.getLength());
            System.out.println("Messaggio ricevuto: " + message);
            socket.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: 04-Socket-UDP/esempi/01.01-unicast-java/udp-receiver.js
/**
 * udp-receiver.js
 * Esempio di ricevitore UDP in JavaScript/Node.js
 * Riceve un messaggio UDP e lo stampa a schermo
 * Versione: 1.0
 * Data: 25/01/26
 * Autore: Filippo Bilardo
 * 
 * node udp-receiver.js
 */

const dgram = require('dgram');

// Crea socket UDP
const socket = dgram.createSocket('udp4');

// Configurazione
const port = 9876;

// Gestione errori
socket.on('error', (err) => {
    console.error('Errore socket:', err);
    socket.close();
    process.exit(1);
});

// Quando il socket è pronto
socket.on('listening', () => {
    const address = socket.address();
    console.log(`In attesa di messaggi su ${address.address}:${address.port}...`);
});

// Quando arriva un messaggio
socket.on('message', (msg, remote) => {
    const message = msg.toString();
    console.log(`Messaggio ricevuto: ${message}`);
    console.log(`Da: ${remote.address}:${remote.port}`);
    
    // Chiude il socket dopo aver ricevuto un messaggio
    socket.close();
});

// Binding sulla porta
socket.bind(port);
